packages/react-devtools-shared/src/devtools/views/Profiler/SnapshotCommitList.js

```diff
@@ -200,15 +207,68 @@ function List({
 
   let tooltipLabel = null;
   if (hoveredCommitIndex !== null) {
-    const commitDuration = commitDurations[hoveredCommitIndex];
-    const commitTime = commitTimes[hoveredCommitIndex];
-    tooltipLabel = `${formatDuration(commitDuration)}ms at ${formatTime(
-      commitTime,
-    )}s`;
+    const {
+      duration,
+      effectDuration,
+      passiveEffectDuration,
+      priorityLevel,
+      timestamp,
+    } = commitData[hoveredCommitIndex];
+
+    // Only some React versions include commit durations.
+    // Show a richer tooltip only for builds that have that info.
+    if (effectDuration > 0 || passiveEffectDuration > 0) {
+      tooltipLabel = (
+        <ul className={styles.TooltipList}>
+          {priorityLevel !== null && (
+            <li className={styles.TooltipListItem}>
+              <label className={styles.TooltipLabel}>Priority</label>
+              <span className={styles.TooltipValue}>{priorityLevel}</span>
+            </li>
+          )}
+          <li className={styles.TooltipListItem}>
+            <label className={styles.TooltipLabel}>Committed at</label>
+            <span className={styles.TooltipValue}>
+              {formatTime(timestamp)}s
+            </span>
+          </li>
+          <li className={styles.TooltipListItem}>
+            <label className={styles.TooltipLabel}>Render duration</label>
+            <span className={styles.TooltipValue}>
+              {formatDuration(duration)}ms
+            </span>
+          </li>
+          {effectDuration !== null && (
+            <li className={styles.TooltipListItem}>
+              <label className={styles.TooltipLabel}>
+                Layout effects duration
+              </label>
+              <span className={styles.TooltipValue}>
+                {formatDuration(effectDuration)}ms
+              </span>
+            </li>
+          )}
+          {passiveEffectDuration !== null && (
+            <li className={styles.TooltipListItem}>
+              <label className={styles.TooltipLabel}>
+                Passive effects duration
+              </label>
+              <span className={styles.TooltipValue}>
+                {formatDuration(passiveEffectDuration)}ms
+              </span>
+            </li>
+          )}
+        </ul>
+      );
+    } else {
+      tooltipLabel = `${formatDuration(duration)}ms at ${formatTime(
+        timestamp,
```

Comment:
Note: When `effectDuration` and `passiveEffectDuration` are present, `totalDuration` includes render and commit measurements. But when only `duration` is present it, doesn't really matter which we use (since `duration === totalDuration`).